bson/src/main-legacy/org/bson/BasicBSONObject.java

```diff
@@ -368,7 +357,7 @@ public boolean equals(final Object o) {
             return false;
         }
 
-        return Arrays.equals(canonicalizeBSONObject(this).encode(), canonicalizeBSONObject(other).encode());
+        return Arrays.equals(getEncoder().encode(canonicalizeBSONObject(this)), getEncoder().encode(canonicalizeBSONObject(other)));
```

Comment:
I could revert this change now.  It's leftover from a refactoring that proved unnecessary.

Reply:
Happy either way.